color/src/serialize.rs

```diff
@@ -136,3 +154,47 @@ impl core::fmt::UpperHex for Rgba8 {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::parse_color;
+
+    #[test]
+    fn specified_to_serialized() {
+        for (specified, expected) in [
+            ("rgb(1,1,1)", "rgb(1, 1, 1)"),
+            // TODO: output rounding? Otherwise the tests should check for approximate equality
+            // (and not string equality) for these conversion cases
+            (
+                "hwb(740deg 20% 30% / 50%)",
+                "rgba(178.5, 93.50008, 50.999996, 0.5)",
+            ),
+            // the next two currently fail, but should succeed (ASCII uppercase codepoints should
+            // be lowercased)
+            // ("ReD", "red"),
+            // ("RgB(1,1,1)", "rgb(1, 1, 1)"),
+            // currently fails, but should succeed (values should be clamped at parse-time)
+            // ("rgb(1.1,1,1)", "rgb(1, 1, 1)"),
+            ("color(srgb 1.0 1.0 1.0)", "color(srgb 1 1 1)"),
+        ] {
```

Comment:
I believe this is the correct serialization as specified by CSS Color 4. To leave case normalization out of the crate, callers wanting case-insensitivity could be instructed to [`str::make_ascii_lowercase`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.make_ascii_lowercase). Otherwise the parser requires some work to get performant insensitive matching.